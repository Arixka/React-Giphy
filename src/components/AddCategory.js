import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    console.log('hangleSubmit', inputValue)
    e.preventDefault();
    if(inputValue.trim().length>2){
      // console.log(setCategories(cat=>[inputValue,...cat]))
      setInputValue('')
    }

  }
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type='text'
        placeholder='Add Category'
        value={inputValue}
        onChange={handleInputValue}
      ></input>
    </form>
  )
}

AddCategory.propTypes= {
  setCategories: PropTypes.func.isRequired
}
