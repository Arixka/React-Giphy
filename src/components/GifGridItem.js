import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title,url}) => {
 // console.log(id, title,url)
  return (
    <div className='card animate__animated animate__fadeIn animate__delay-2s'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}