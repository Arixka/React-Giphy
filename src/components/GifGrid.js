import React from 'react'
import PropTypes from 'prop-types'
import {useFetchGifs} from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifs(category)

  return (
    <>
      {loading && <p className="animate__animated animate__flash animate__delay-2s">Loading...</p>}

      <h3 className="animate__animated animate__fadeIn animate__delay-2s">
        {category}
      </h3>
      <div className='card-grid'>
        {images.map((img) =>
          <GifGridItem
            key={img.id}
            {...img} />//...img mandamos cada imagen con sus propiedades 
        )}

      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}