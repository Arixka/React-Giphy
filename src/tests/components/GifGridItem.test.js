import React from 'react';
import { shallow } from 'enzyme'
import {GifGridItem} from '../../components/GifGridItem'

describe('pruebas con gifGridItem', () => {

  const title = 'titulo'
  const url = 'http://loquesea'
  let wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('Debe renderizar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('Debe tener un parrafo con el titulo', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })
  test('Debe tener una imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img')
    expect(img.prop('alt')).toBe('titulo')
    expect(img.prop('src')).toBe('http://loquesea')
  })

  test('Debe tener la clase animate_fadeIn',()=>{
    const div = wrapper.find('div')
    expect(div.prop('className')).toContain('animate__fadeIn')
  })

})