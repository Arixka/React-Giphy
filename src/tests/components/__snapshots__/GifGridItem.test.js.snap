// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con gifGridItem Debe renderizar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="titulo"
    src="http://loquesea"
  />
  <p>
    titulo
  </p>
</div>
`;
